### PR TITLE
Update bind loop support for namespaced classes and plans

### PR DESCRIPTION
**Pull request by jmrodri**

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -637,29 +637,81 @@ func (c *controller) getClusterServiceBrokerForServiceBinding(instance *v1beta1.
...
 
-	clientConfig := NewClientConfigurationForBroker(broker.ObjectMeta, &broker.Spec.CommonServiceBrokerSpec, authConfig)
+	var brokerClient osb.Client
 
-	glog.V(4).Infof("Creating client for ClusterServiceBroker %v, URL: %v", broker.Name, broker.Spec.URL)
-	brokerClient, err := c.brokerClientCreateFunc(clientConfig)
-	if err != nil {
-		return nil, err
+	if instance.Spec.ClusterServiceClassSpecified() {
+
```
Comment (n3wscott) [inline pkg/controller/controller.go]: nit: what is with all this extra space in the new block?

File: pkg/controller/controller_binding.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 
 	"bytes"
```
Comment (n3wscott) [inline pkg/controller/controller_binding.go]: bytes should go up by net and fmt

File: pkg/controller/controller_binding.go [hunk 3]
```diff
@@ -198,36 +199,84 @@ func (c *controller) reconcileServiceBindingAdd(binding *v1beta1.ServiceBinding)
...
+	var inProgressProperties *v1beta1.ServiceBindingPropertiesState
 
-	serviceClass, servicePlan, brokerName, brokerClient, err := c.getClusterServiceClassPlanAndClusterServiceBrokerForServiceBinding(instance, binding)
-	if err != nil {
-		return c.handleServiceBindingReconciliationError(binding, err)
-	}
+	if instance.Spec.ClusterServiceClassSpecified() {
+		if instance.Spec.ClusterServiceClassRef == nil || instance.Spec.ClusterServicePlanRef == nil {
+			// retry later
+			msg := fmt.Sprintf(`Binding cannot begin because ClusterServiceClass and ClusterServicePlan references for %s have not been resolved yet`, pretty.ServiceInstanceName(instance))
```
Comment (n3wscott) [inline pkg/controller/controller_binding.go]: pretty print the msg?

File: pkg/controller/controller.go (hunk 1 above)
Comment (jmrodri) [reply pkg/controller/controller.go]: I can fix it if you want, it made it easier for me to read and see the if blocks especially when I was moving things around. But now that it's done I'd be happy to remove it if required.

File: pkg/controller/controller_binding.go (hunk 2 above)
Comment (jmrodri) [reply pkg/controller/controller_binding.go]: Wow, not sure why goimports didn't move that for me. It's usually really good at taking of that. I'll fix that.

File: pkg/controller/controller_binding.go (hunk 3 above)
Comment (jmrodri) [reply pkg/controller/controller_binding.go]: @n3wscott I'm not quite sure what you mean by that? :) not trying to obtuse, but clearly there is a pretty printer used that I missed, is there a good example of it being used elsewhere?

File: pkg/controller/controller.go (hunk 1 above)
Comment (jmrodri) [reply pkg/controller/controller.go]: I did it in controller_binding.go as well. Let me know if you really want me to fix this nit or not.

**Pull request merged**